api/src/services/listing.service.ts

```diff
@@ -1139,6 +1140,38 @@ export class ListingService implements OnModuleInit {
   async update(dto: ListingUpdate, requestingUser: User): Promise<Listing> {
     const storedListing = await this.findOrThrow(dto.id, ListingViews.details);
 
+    // block date changes on active listing for non admins
+    if (
+      requestingUser?.userRoles?.isAdmin === false &&
+      dto.status === 'active'
+    ) {
+      const lotteryEvent = dto.listingEvents?.find(
+        (event) => event?.type === ListingEventsTypeEnum.publicLottery,
+      );
+      const storedLotteryEvent = storedListing.listingEvents?.find(
+        (event) => event?.type === ListingEventsTypeEnum.publicLottery,
+      );
+      if (
```

Comment:
@KrissDrawing This `if` block is a little hard for me to parse…maybe it could use some intermediate variables/functions to make the logic clearer to follow? Or some additional comments?